File: solutions/elixir/114-invert-binary-tree.ex
# Problem 114: Invert Binary Tree (LeetCode 226)
# Difficulty: Easy
# Language: Elixir
# 
defmodule InvertTree do
  def invert(nil), do: nil
  def invert({val, left, right}),
    do: {val, invert(right), invert(left)}
end

File: solutions/typescript/114-invert-binary-tree.ts
/**
 * Problem 114: Invert Binary Tree (LeetCode 226)
 * Difficulty: Easy
 * Language: TypeScript
 */
type TreeNode = { val: number; left: TreeNode | null; right: TreeNode | null };
function invertTree(root: TreeNode | null): TreeNode | null {
  if (root === null) return null;
  return {
    val: root.val,
    left: invertTree(root.right),
    right: invertTree(root.left)
  };
}
